src/programs/state/massachusetts/snap.js

```diff
@@ -1,14 +1,443 @@
+  //   var result = new Result({
+  //     result: 'incomplete',
+  //     details: 'Form incomplete',
+  //     data: { missingProps: missingProps }
+  //   });
+  //   return result;
+  // }
+
+  var finalResult = null;
+  var grossIncomeTestResult   = getGrossIncomeTestResult(client, timeframe);
+  var netIncomeTestResult     = getNetIncomeTestResult(client, timeframe);
+  var maxSnapAllotment        = getMaxSnapAllotment(client, timeframe);
+  var thirtyPercentNetIncome  = getThirtyPercentNetIncome(client, timeframe);
+  var maxClientAllotment      = maxSnapAllotment - thirtyPercentNetIncome;
+
+  if (grossIncomeTestResult === true &&  netIncomeTestResult === true) {
+
+    if ( maxClientAllotment <= data.smallHouseholdMinimumGrant ) {
+      if (client[timeframe + 'HouseholdSize'] <= data.minHouseholdSize) {
+        finalResult = data.smallHouseholdMinimumGrant;
+      } else {
+        finalResult = 0;
+      }
+    } else {
+      finalResult = maxClientAllotment;
+    }
+  } else {
+    finalResult = 0;
+  }
+
+  var result = new Result({
+      result: 'good',
+      details: 'IT IS AWESOME!!!',
+      benefitValue: finalResult
+    });
+  return result;
+}; // End getSNAPBenefits()
+
+//Still needed, commented out requiredProps[] for removing warning
+// const requiredProps = [
+//   'currentDisabledOrElderlyMember',
+//   'futureEarnedIncomeMonthly',
+//   'currentTAFDCMonthly',
+//   'currentSSIMonthly',
+//   'currentSSDIMonthly',
+//   'currentChildSupportInMonthly',
+//   'currentUnemploymentMonthly',
+//   'currentWorkersCompMonthly',
+//   'currentPensionMonthly',
+//   'currentSocialSecurityMonthly',
+//   'currentAlimonyMonthly',
+//   'currentOtherIncomeMonthly',
+//   'currentHouseoldSize',
+//   'currentChildSupportPaidOutMonthly',
+//   'currentChildDirectCareCostsMonthly',
+//   'currentChildBeforeAndAfterSchoolCareCostsMonthly',
+//   'currentChildTransportationCostsMonthly',
+//   'currentChildOtherCareCostsMonthly',
+//   'currentAdultDirectCareCostsMonthly',
+//   'currentAdultTransportationCostsMonthly',
+//   'currentAdultOtherCareCostsMonthly',
+//   'currentDisabledMedicalCostsMonthly',
+//   'currentHomeless',
+//   'currentMortgageMonthly',
+//   'currentHousingInsuranceMonthly',
+//   'currentPropertyTaxMonthly',
+//   'currentPaidUtilities',
+//   'currentClimateControl',
+//   'currentNonHeatElectricity',
+//   'currentPhone',
+//
+//
+//   // 'currentHomeowner',
+//   /* All question marks in spreadsheet
+//   'currentIncomeExclusionsMonthly',
+//   'currentDisabledAssistanceMonthly',
+//   'currentOtherMedicalCostsMonthly',
+//   'currentRentShareMonthly',
+//   'currentContractRentMonthly',
+//   'currentRentMonthly'
+//   */
+// ];
+
+const getAllowance = function(client, timeframe, data, baseRate ) {
+  if ( client[timeframe + 'HouseholdSize'] > 8  ) {
+    return data[8] + (client[timeframe + 'HouseholdSize'] - 8) * baseRate;
+  } else {
```

Comment:
`getMaxIntKey()` should probably be an exported method, since it seems that `baseRate` changes a lot. @knod I want to hear your thoughts on how to use the helper methods here

Reply:
@shp54: He can actually just use `getMonthlyLimitBySize()`. It's exactly what it's built for. In fact, it would be perfect for both of these. It's part of why I didn't export `getMaxIntKey()` - the other functions are already taking care of what needs to be done. If we find we have more need of `getMaxIntKey()`, we'll set up a file for more utility-type stuff and export it there (it's not like it really has much to do with getting government data other than being useful to the process).